### PR TITLE
docs: Documentation update for OpenTelemetry

### PR DESCRIPTION
**Pull request by gkevinzheng**
- Added page with more info about automatic trace/span ID extraction
- Updated Automatic Metadata Detection section with reference to new page

Sphinx preview:
image
image

### CODE REVIEW

File: docs/auto-trace-span-extraction.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+-------------------------
+
+If you have the OpenTelemetry SDK package installed and are logging from within an active OpenTelemetry span, that log entry will automatically
+have the `trace`, `span_id`, and `trace_sampled` fields populated with metadata from that span. More information about OpenTelemetry can be found 
+`here <https://opentelemetry.io/docs/languages/python/>`_.
+
+HTTP headers
+------------
+
+Another possible method of automatic `trace`/ `span_id` is via extraction from HTTP headers. This feature requires a :doc:`supported Python web framework </web-framework-integration>`.
```
Comment (cindy-peng) [inline docs/auto-trace-span-extraction.rst:17]: Do we want to mention that this is prioritized after OpenTelemetry?

File: docs/auto-trace-span-extraction.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Automatic Trace/Span ID Extraction
+==================================
+
+The Google Cloud Logging library can automatically populate `LogEntry fields <https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry>`_
+`trace`, `span_id`, and`trace_sampled` via OpenTelemetry integration, or extracting header information from an HTTP request.
```
Comment (cindy-peng) [inline docs/auto-trace-span-extraction.rst:5]: Somehow and`trace_sampled` is not populated correctly in the screenshot.

File: docs/auto-trace-span-extraction.rst (hunk 1 above)
Comment (mtabasko) [reply docs/auto-trace-span-extraction.rst:17]: I think that's probably a good idea.

**Review by mtabasko — CHANGES_REQUESTED** [review]

File: docs/auto-trace-span-extraction.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    All `LogEntry fields`_ populated :ref:`manually<Manual-Metadata>` will override those populated via methods referred to in this
+    section.
+
+The Google Cloud Logging library can automatically populate `LogEntry fields`_
+`trace`, `span_id`, and `trace_sampled` via OpenTelemetry integration, or extracting header information from an HTTP request.
+
+OpenTelemetry Integration
+-------------------------
+
+If you have the OpenTelemetry SDK package installed and are logging from within an active OpenTelemetry span, that log entry will automatically
```
Comment (aabmass) [inline docs/auto-trace-span-extraction.rst:14]: nit the SDK is not technically necessary here. Does this suggestion make sense to you? I think it's a little more precise
```suggestion
If you are using OpenTelemetry and there is an active span in the OpenTelemetry Context, that log entry will automatically
```

File: docs/auto-trace-span-extraction.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+have the `trace`, `span_id`, and `trace_sampled` fields populated with metadata from that span. More information about OpenTelemetry can be found
```
Comment (aabmass) [inline docs/auto-trace-span-extraction.rst:15]: nit

```suggestion
have the `trace`, `span_id`, and `trace_sampled` fields populated from that span. More information about OpenTelemetry can be found 
```

**Review by aabmass — APPROVED** [review]

**Review by cindy-peng — APPROVED** [review]

**Review by mtabasko — APPROVED** [review]

**Pull request merged**